### ISSUE TITLE
Pubsub acks are unsuccessful

### ISSUE THREAD

**Issue by stetra** [googleapis/google-cloud-java#3567]
When I use the streaming pull subscriber through the `MessageReceiver` interface, it seems acks usually expire, and even if they don't expire, the number of backlogged messages never changes (num_undelivered_messages in Stackdriver). This only seems to happen if messages take a while to be processed, i.e., more than a few minutes.

It's strange that acks are expired in the first place because I do see that the client is sending modify ack deadline requests (mod_ack_deadline_message_operation_count in Stackdriver). The subscription itself has an ack deadline of 10 minutes so these messages are processed well within the deadline.

Even when acks are successful (i.e., non-expired), num_undelivered_messages is unaffected in Stackdriver. I do see the previously acked messages being redelivered as well.

Another thing I noticed that even though the client is performing streaming pulls, in Stackdriver I see metric data coming in for `pull_ack_message_operation_count` but none for `streaming_pull_ack_message_operation_count`. Similarly I see results for `mod_ack_deadline_message_operation_count` but not for `streaming_pull_mod_ack_deadline_message_operation_count`. Is this expected?

I am using version 1.37.1 of the google-cloud-pubsub library. I have uploaded my code here: https://gist.github.com/stetra/d757fed41cb67d4a73dd7487ca4d452e

The following Stackdriver graphs show that successful acks are apparently happening, yet the backlog is not decreasing.
![acks](https://user-images.githubusercontent.com/35081873/44225145-80c2db00-a141-11e8-970a-d503163269bf.png)
![num_undelivered_messages](https://user-images.githubusercontent.com/35081873/44225144-80c2db00-a141-11e8-8619-7d485869ef23.png)
![modacks](https://user-images.githubusercontent.com/35081873/44225143-80c2db00-a141-11e8-8dc2-9effe00daf8d.png)

In summary, these are my questions:

1. Despite successful acks occurring, why is the number of unacknowledged messages not decreasing?

2. Why are expired acks occurring in the first place?

3. Why is the client performing streaming pulls with non-streaming acks and modacks?

**Comment by stetra** [comment]
Thanks for the reply. I did not realize that expired acks behaved that way. Is there documentation for those metric statuses in Stackdriver? E.g., for pull_ack_message_operation_count there's expired and success, but perhaps there's more statuses that I'm not aware of.

The above screenshots are from August 16 around 10 AM Pacific Time, but the issue is still happening at the moment (August 20, ~2:45 PM Pacific Time). The subscription is `d-steven-long-sub` on project `lofty-outcome-860`.

Screenshots:
![num_undelivered_messages2](https://user-images.githubusercontent.com/35081873/44368429-a0cb0500-a487-11e8-8a0a-84d65340a796.png)
![acks2](https://user-images.githubusercontent.com/35081873/44368428-a0cb0500-a487-11e8-9e56-21a049b2c5fa.png)

Please let me know if you need anything else.

**Comment by pongad** [comment]
@stetra I contacted the pubsub team with your info. I'll keep you posted.

**Comment by kimkyung-goog** [comment]
Hi @stetra, I am trying to debug this issue. 
When you observe this behavior, what was the rate of publishing messages? how many messages per second were being published? 
Also, do you observe this issue whenever you run your Repro.java, or does it happen only occasionally?

**Comment by anilmuppalla** [comment]
@pongad you mentioned bringing polling pull back, do you have an ETA on that?

**Comment by pongad** [comment]
I don't want to promise anything right now, but I'll try to work on it over the next couple of weeks if my availability opens up.

**Comment by matthewrj** [comment]
I am seeing similar behaviour where I get a lot of expired responses to acks and the number of undelivered messages stops going down. It seems to happen in 5-20 minute intervals. I am using the java client library version 1.45.0.

My subscriber is created as follows:
```
Subscriber
  .newBuilder(subscriptionName, receiver)
  .setSystemExecutorProvider(
    FixedExecutorProvider.create(Executors.newScheduledThreadPool(8)))
  .setExecutorProvider(FixedExecutorProvider.create(
    Executors.newScheduledThreadPool(128)))
  .setFlowControlSettings(
    FlowControlSettings
      .newBuilder()
      .setMaxOutstandingElementCount(128)
      .build())
  .build()
```

![expired_acks](https://user-images.githubusercontent.com/11561974/46184649-a808ec80-c32a-11e8-9568-f27a9560be62.png)
![undelivered_messages](https://user-images.githubusercontent.com/11561974/46184655-ab9c7380-c32a-11e8-8d38-492b64b613cd.png)

**Comment by pongad** [comment]
#3743  might help. Let's see where that gets us first.

**Comment by matthewrj** [comment]
The processing of messages is also slow in our case. Handing the processing off to another thread pool and keeping the message receiver lightweight appears to have fixed the issue.

**Comment by QuestionAndAnswer** [comment]
I'm also seeing such behavior that @stetra faced, but now on node.js environment. Therefore it may be not related to exactly java implementation of the library. 

Here is the link on the question on SO, in case if this might help.

https://stackoverflow.com/questions/54597310/google-cloud-pubsub-not-ack-messages

**Comment by ajaaym** [comment]
@stetra This issue will be resolved once we bring back polling pull. Closing this issue and we will track in #3500
